runtime/lua/vim/lsp/buf.lua

```diff
@@ -809,9 +808,11 @@ function M.code_action(options)
   ---@type table<integer, vim.lsp.CodeActionResultEntry>
   local results = {}
 
+  --- Handler for textDocument/codeAction
   ---@param err? lsp.ResponseError
   ---@param result? (lsp.Command|lsp.CodeAction)[]
   ---@param ctx lsp.HandlerContext
+  ---@type vim.lsp.ResponseHandler
```

Comment:
I don't think it's a good idea to use `@type` on functions with `@param`, etc

Reply:
I agree. But `@param`'s are needed because they provide the most specific and useful type information, and I wanted to make it clear whether each handler function is either request, response, or notification handler. Any better idea? Probably somewhere in the docstring without annotations? Or we could simply drop that, which would be not necessarily needed.

Reply:
If you have to, I would use a different container for requests, responses and notifications and merge them as M at the end of the file.